cmake/std/atdm/test/unit_tests/set_build_options_unit_tests.sh

```diff
@@ -0,0 +1,484 @@
+#!/bin/bash
+
+CURRENT_SCRIPTS_DIR=`echo $BASH_SOURCE | sed "s/\(.*\)\/.*\.sh/\1/g"`
+ATDM_CONFIG_SCRIPT_DIR=`readlink -f ${CURRENT_SCRIPTS_DIR}/../..`
+
+#
+# Test compiler parsing
+#
+
+
+testAllDefaults() {
+  ATDM_CONFIG_BUILD_NAME=default
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} DEFAULT
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_KOKKOS_ARCH} DEFAULT
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_BUILD_TYPE} DEBUG
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_SHARED_LIBS} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_NODE_TYPE} SERIAL
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_USE_OPENMP} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_USE_CUDA} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_USE_PTHREADS} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_CUDA_RDC} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_FPIC} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_FPIC} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPLEX} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_SHARED_LIBS} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_PT_PACKAGES} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_FINISHED_SET_BUILD_OPTIONS} 1
+}
+
+
+testCompilerClangAndDefaults() {
+  ATDM_CONFIG_BUILD_NAME=clang
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} CLANG
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_KOKKOS_ARCH} DEFAULT
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_BUILD_TYPE} DEBUG
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_SHARED_LIBS} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_NODE_TYPE} SERIAL
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_USE_OPENMP} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_USE_CUDA} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_USE_PTHREADS} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_CUDA_RDC} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_FPIC} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_FPIC} OFF
```

Comment:
Is this a duplicate of line 44?

Reply:
Correct again.  That was a fast copy and paste error.  I will remove the extra one.